src/azure-cli/azure/cli/command_modules/resource/tests/latest/test_resource.py

```diff
@@ -1622,31 +1622,49 @@ def test_subscription_level_what_if_ts(self, resource_group, resource_group_loca
 
 
 class DeploymentScriptsTest(ScenarioTest):
-    @ResourceGroupPreparer(name_prefix='cli_test_deployment_scripts')
+    @ResourceGroupPreparer(name_prefix='cli_test_deployment_scripts', location='brazilsouth')
     def test_list_all_deployment_scripts(self, resource_group):
         curr_dir = os.path.dirname(os.path.realpath(__file__))
         self.kwargs.update({
-            'deployment_script_name': self.create_random_name('script', 20),
-            'deployment_name': self.create_random_name('ds', 20),
+            'deployment_script_name': self.create_random_name('script2019', 20),
+            'deployment_name': self.create_random_name('ds2019', 20),
             'resource_group': resource_group,
+            'api_version': '2019-10-01-preview',
             'template_file': os.path.join(curr_dir, 'deployment-scripts-deploy.json').replace('\\', '\\\\'),
         })
 
         count = 0
         self.cmd('deployment-scripts list',
                  checks=self.check("length([?name=='{deployment_script_name}'])", count))
 
-        self.cmd('deployment group create -g {resource_group} -n {deployment_name} --template-file "{template_file}" --parameters scriptName={deployment_script_name}', checks=[
+        self.cmd('deployment group create -g {resource_group} -n {deployment_name} --template-file "{template_file}" --parameters scriptName={deployment_script_name} --parameters apiVersion="{api_version}"', checks=[
```

Comment:
May I ask why we keep the test for the historical api-version?

Reply:
Just to make sure the previous version works, main reason is that in the previous version the identity property is required and in the latest is optional, so just keeping a test to ensure both approaches work, not sure if is an accepted approach or not, happy to remove the historical api-version.

Reply:
If the user will still call the old api-version in new CLI version, then I think this test can be retained.
If the user has barely used the old api-version in new CLI version, we can delete it~

Reply:
Since we can not specify the api version, I'll delete this. Thanks @zhoxing-ms for the feedback, will provide a new iteration.